app/src/main/java/com/mealsmadeeasy/ui/login/LoginActivity.kt

```diff
@@ -0,0 +1,92 @@
+    private lateinit var signInClient: GoogleSignInClient
+
+    companion object {
+        fun newIntent(context: Context) = Intent(context, LoginActivity::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        findViewById<SignInButton>(R.id.sign_in_button).setOnClickListener{signIn()}
+
+        // Configure sign-in to request the user's ID, email address, and basic
+        // profile. ID and basic profile are included in DEFAULT_SIGN_IN.
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(getString(R.string.default_web_client_id)) //OAuth 2.0 client ID
+                .requestEmail()
+                .build()
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        signInClient = GoogleSignIn.getClient(this, gso)
+        auth = FirebaseAuth.getInstance()
+    }
+
+    public override fun onStart() {
+        super.onStart()
+        // Check if user is signed in (non-null) and update UI accordingly.
+//        val currentUser = auth.currentUser
+//        if (currentUser != null) {
+//            startActivity(HomeActivity.newIntent(this))
+//        }
+    }
+
+    private fun signIn() {
+        val signInIntent = signInClient.signInIntent
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        if (requestCode == RC_SIGN_IN) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            try {
+                // Google Sign In was successful, authenticate with Firebase
+                val account = task.getResult(ApiException::class.java)
+                firebaseAuthWithGoogle(account)
+            } catch (e: ApiException) {
+                // Google Sign In failed, display a message to the user.
+                Snackbar.make(findViewById(R.id.activity_login), "@string/google_login_fail", Snackbar.LENGTH_SHORT).show()
+            }
+
+        }
+    }
+
+    private fun firebaseAuthWithGoogle(acct: GoogleSignInAccount) {
+        val credential = GoogleAuthProvider.getCredential(acct.idToken, null)
+        auth.signInWithCredential(credential)
+                .addOnCompleteListener(this) { task ->
+                    if (task.isSuccessful) {
+                        // Sign in success
+                        startActivity(HomeActivity.newIntent(this))
+                    } else {
+                        // If sign in fails, display a message to the user.
+                        Snackbar.make(findViewById(R.id.activity_login), "@string/auth_fail", Snackbar.LENGTH_SHORT).show()
```

Comment:
This should be `R.string.auth_fail`. Not `"@String/auth_fail"` (That only works in XML).